openraft/src/core/mod.rs

```diff
@@ -945,9 +946,11 @@ impl<'a, C: RaftTypeConfig, N: RaftNetworkFactory<C>, S: RaftStorage<C>> LeaderS
                 blocking,
                 turn_to_learner,
                 tx,
-            } => {
-                self.change_membership(members, blocking, turn_to_learner, tx).await?;
-            }
+            } => match members {
+                ChangeMembers::Replace(c) => self.change_membership(c, blocking, turn_to_learner, tx).await?,
+                ChangeMembers::Add(c) => self.add_members(c, blocking, tx).await?,
+                ChangeMembers::Remove(c) => self.remove_members(c, blocking, turn_to_learner, tx).await?,
+            },
```

Comment:
It looks like by moving this `match` into `self.change_membership()` it will get rid of about 50 lines of code.